diff --git a/dwm.c b/dwm.c
index e62f6a8..58233db 100644
--- a/dwm.c
+++ b/dwm.c
@@ -973,7 +973,7 @@ drawbar(Monitor *m)
 	/* draw status first so it can be overdrawn by tags later */
 	if (m == selmon) { /* status is only drawn on selected monitor */
 		drw_setscheme(drw, scheme[SchemeNorm]);
-		tw = TEXTW(stext) - lrpad + 2; /* 2px padding */
+		tw = TEXTW(stext) - lrpad / 2 + 2; /* 2px padding */
 		drw_text(drw, m->ww - tw - stw, 0, tw, bh, lrpad / 2 - 2, stext, 0);
 	}
 
